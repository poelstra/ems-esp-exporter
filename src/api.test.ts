import { expect, it } from "@jest/globals";
import { getExampleSystem_3_5, getExampleSystem_3_7 } from "./test/examples";

it("parses system info v3.5", async () => {
    expect(getExampleSystem_3_5()).toMatchSnapshot();
});

it("parses system info v3.7", async () => {
    expect(await getExampleSystem_3_7()).toMatchSnapshot();
});
