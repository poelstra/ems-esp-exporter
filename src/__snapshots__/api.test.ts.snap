// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses original system info 1`] = `
{
  "apiInfo": {
    "apiCalls": 137014,
    "apiFails": 2,
  },
  "busInfo": {
    "incompleteTelegrams": 5,
    "protocol": "Buderus",
    "readsFailed": 0,
    "rxLineQuality": 100,
    "rxTelegramsReceived": 266780,
    "status": "connected",
    "txLineQuality": 100,
    "txReads": 35526,
    "txWrites": 7978,
    "writesFailed": 0,
  },
  "devices": [
    {
      "deviceId": 8,
      "entities": 68,
      "handlersFetched": [
        20,
        22,
        51,
      ],
      "handlersIgnored": undefined,
      "handlersPending": [
        191,
        194,
        26,
        53,
        38,
        42,
        209,
        227,
        228,
        229,
        230,
        233,
        234,
      ],
      "handlersReceived": [
        16,
        17,
        21,
        28,
        24,
        25,
        52,
      ],
      "name": "Topline/GB162",
      "productId": 115,
      "type": "boiler",
      "version": "03.06",
    },
    {
      "deviceId": 9,
      "entities": 0,
      "handlersFetched": undefined,
      "handlersIgnored": undefined,
      "handlersPending": undefined,
      "handlersReceived": undefined,
      "name": "BC10",
      "productId": 114,
      "type": "controller",
      "version": "01.03",
    },
  ],
  "mqttInfo": {
    "base": "ems-esp",
    "cleanSession": true,
    "clientId": "ems-esp",
    "discoveryPrefix": "homeassistant",
    "enabled": true,
    "entityFormat": 0,
    "haEnabled": true,
    "keepAlive": 60,
    "mqttQos": 0,
    "mqttRetain": false,
    "nestedFormat": 1,
    "publish2command": false,
    "publishSingle": false,
    "publishTimeBoiler": 10,
    "publishTimeHeartbeat": 60,
    "publishTimeMixer": 10,
    "publishTimeOther": 10,
    "publishTimeSensor": 10,
    "publishTimeSolar": 10,
    "publishTimeThermostat": 10,
    "sendResponse": false,
    "status": "connected",
  },
  "networkInfo": {
    "ApProvisionMode": "disconnected",
    "ApSecurity": "wpa2",
    "ApSsid": "ems-esp",
    "RSSI": -57,
    "corsOrigin": undefined,
    "disableSleep": false,
    "enableCors": false,
    "enableIpv6": false,
    "enableMdns": true,
    "hostname": "ems-esp",
    "ipv4Address": "192.168.3.15/255.255.255.0",
    "ipv4Gateway": "192.168.3.1",
    "ipv4Nameserver": "192.168.3.1",
    "lowBandwidth": false,
    "network": "WiFi",
    "staticIpConfig": false,
  },
  "ntpInfo": {
    "enabled": true,
    "server": "nl.pool.ntp.org",
    "status": "connected",
    "tzLabel": "Europe/Amsterdam",
  },
  "otaInfo": {
    "enabled": false,
    "port": 8266,
  },
  "sensorInfo": {
    "analogSensorFails": 0,
    "analogSensorReads": 0,
    "analogSensors": 0,
    "temperatureSensorFails": 0,
    "temperatureSensorReads": 0,
    "temperatureSensors": 0,
  },
  "settings": {
    "analogEnabled": true,
    "boardProfile": "S32",
    "boolDashboard": 1,
    "boolFormat": 1,
    "dallasParasite": false,
    "emsBusId": 11,
    "enumFormat": 1,
    "fahrenheit": false,
    "hideLed": false,
    "locale": "en",
    "maxWebLogBuffer": 50,
    "notokenApi": false,
    "readonlyMode": false,
    "showerAlert": false,
    "showerTimer": false,
    "telnetEnabled": true,
    "txMode": 1,
    "webLogBuffer": 50,
  },
  "syslogInfo": {
    "enabled": false,
  },
  "systemInfo": {
    "appFreeKb": 6152,
    "memFreeKb": 117,
    "memMaxAllocKb": 65,
    "platform": "ESP32",
    "uptimeSeconds": 645036,
    "version": "3.5.1-dev.0",
  },
}
`;
